### DISCUSSION TITLE
MCDM and visualization assumption.

### DISCUSSION THREAD

**Discussion by muazhari** [anyoptimization/pymoo#614] [Q&A] [answered]
In my optimization use case, one of the objectives is maximization.
To be calculated by Pymoo, the maximization objectives must be multiplied by `-1` (converted to minimization).
Based on [this](https://math.stackexchange.com/questions/2928015/clarification-about-the-relation-between-maximization-and-minimization-of-object), after the `minimize()` function executed, the `result.F` which are maximization need to be multiplied by `-1` again to back to the original.
Then, I want to supply the result to MCDM and the visualization features in Pymoo.
Does MCDM and visualization always assume the input of objectives is minimization?

Also, can you provide the normalization/preprocess function to be able easily used for MCDM and minimization in Pymoo?
Unknowingly, I do not have any knowledge to do it right. 
Especially when faced with edge cases like what if there is a division of zero?
You can directly correct the formula below in this discussion because I need it as soon as possible. 

```python
nF = (res.F - approx_ideal) / (approx_nadir - approx_ideal)
```

```python
i = decomp.do(nF, 1 / weights).argmin()
```

Thanks in advance!

**Comment by blankjul** [comment, marked as answer]
The entire framework assumes minimization. So does MCDM. 

I recommend fixing the edge case by using an `eps` value (this is probably the most practical fix), e.g.
```
max(weights, 1e-16)
```
